### PR TITLE
[ODPR-1673] - Refactor code related to sending email for R1.2

### PR DESCRIPTION
**Pull request by georgievh**

### CODE REVIEW

File: app/connectors/EmailConnector.scala [hunk 1]
```diff
@@ -32,20 +33,20 @@ import scala.util.control.NonFatal
...
       .withBody(Json.toJson(sendEmailRequest))
       .execute[HttpResponse]
       .flatMap { response =>
         response.status match {
-          case ACCEPTED => Future.successful(true)
+          case ACCEPTED => Future.successful(Done)
           case status =>
             logger.warn(s"Send email failed with status: $status")
-            Future.successful(false)
+            Future.successful(Done)
```
Comment (jassalrichy) [inline app/connectors/EmailConnector.scala:45]: Test for when any status other than 202 is returned.

File: app/services/EmailService.scala [hunk 2]
```diff
@@ -19,25 +19,52 @@ package services
...
+                                         addNotificationRequest: UpdatePlatformOperatorRequest)
+                                        (implicit hc: HeaderCarrier): Future[Done] = {
+    sendEmail(AddedReportingNotificationRequest.build(userAnswers, subscriptionInfo))
+    sendEmail(AddedAsReportingNotificationRequest.build(userAnswers, addNotificationRequest))
   }
 
+  private def sendEmail(requestBuild: EitherNec[Query, SendEmailRequest])
+                       (implicit hc: HeaderCarrier): Future[Done] = requestBuild.fold(
+    errors => {
+      logger.warn(s"Unable to send email, path(s) missing:" +
```
Comment (jassalrichy) [inline app/services/EmailService.scala:64]: Would be useful to output the type of email that failed.

File: app/controllers/update/RemovePlatformOperatorController.scala [hunk 3]
```diff
@@ -53,41 +52,27 @@ class RemovePlatformOperatorController @Inject()(
...
-              _                 <- emailService.sendEmail(RemovedPlatformOperatorRequest.build(request.userAnswers, subscriptionInfo))
-              _                 <- emailService.sendEmail(RemovedAsPlatformOperatorRequest.build(request.userAnswers))
-              _                 <- auditService.sendAudit[RemovePlatformOperatorAuditEventModel](
-                                  RemovePlatformOperatorAuditEventModel(businessName, operatorId).toAuditModel)
-            } yield Redirect(routes.PlatformOperatorRemovedController.onPageLoad(operatorId))
-          } else {
-            Future.successful(Redirect(routes.PlatformOperatorController.onPageLoad(operatorId)))
-          }
+      formProvider(businessName).bindFromRequest().fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, operatorId, businessName))),
```
Comment (jassalrichy) [inline app/controllers/update/RemovePlatformOperatorController.scala:62]: missing test for formWithErrors.

File: app/connectors/EmailConnector.scala (hunk 1 above)
Comment (georgievh) [reply app/connectors/EmailConnector.scala:45]: There is a test for this already as discussed.

File: app/controllers/update/RemovePlatformOperatorController.scala (hunk 3 above)
Comment (georgievh) [reply app/controllers/update/RemovePlatformOperatorController.scala:62]: This is a missing test from when the page was implemented initially. I have added a test to cover that now.

**Review by jassalrichy — APPROVED** [review]

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**